app/services/projects/transfer_service.rb

```diff
@@ -34,19 +35,27 @@ def execute(new_namespace) # rubocop:disable Metrics/AbcSize
 
     private
 
-    attr_reader :new_namespace
-
-    def transfer(project)
-      if Namespaces::ProjectNamespace.where(parent_id: @new_namespace.id).exists?(['path = ? or name = ?',
-                                                                                   project.path, project.name])
+    def transfer(project) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+      if Namespaces::ProjectNamespace.where(parent_id: new_namespace.id).exists?(['path = ? or name = ?',
+                                                                                  project.path, project.name])
         raise TransferError, I18n.t('services.projects.transfer.namespace_project_exists')
       end
 
       project_ancestor_member_user_ids = Member.for_namespace_and_ancestors(project.namespace).select(:user_id)
-      new_namespace_member_ids = Member.for_namespace_and_ancestors(@new_namespace)
+      new_namespace_member_ids = Member.for_namespace_and_ancestors(new_namespace)
                                        .where(user_id: project_ancestor_member_user_ids).select(&:id)
 
-      project.namespace.update(parent_id: @new_namespace.id)
+      project.namespace.update(parent_id: new_namespace.id)
+
+      project.namespace.create_activity action: :transfer,
+                                        owner: current_user,
+                                        parameters:
+                                        {
+                                          project_name: project.name,
```

Comment:
**nitpick (blocking):** Can you remove `project_name` as I don't think it is necessary (e.g. you are not using it in the translations).

Can you change the value of the `old_namespace` and `new_namespace` to store the `puid` instead of the `name`.

Reply:
Updated in 032cb1c08ee993dee5d18b74ef198b03bafe07b3